packages/block-library/src/post-author/edit.js

```diff
@@ -1,31 +1,255 @@
+import {
+	AlignmentToolbar,
+	BlockControls,
+	FontSizePicker,
+	InspectorControls,
+	RichText,
+	__experimentalUseColors,
+	withFontSizes,
+} from '@wordpress/block-editor';
+import { PanelBody, SelectControl, ToggleControl } from '@wordpress/components';
 import { useSelect } from '@wordpress/data';
-import { sprintf, __ } from '@wordpress/i18n';
-
-function PostAuthorDisplay() {
-	const [ authorId ] = useEntityProp( 'postType', 'post', 'author' );
-	const author = useSelect(
-		( select ) =>
-			select( 'core' ).getEntityRecord( 'root', 'user', authorId ),
-		[ authorId ]
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockColorsStyleSelector from './block-colors-selector';
+
+const DEFAULT_AVATAR_SIZE = 24;
+
+function PostAuthorDisplay( { props, postAuthorId } ) {
+	const ref = useRef();
+
+	const [ authorId, setAuthorId ] = useState( props.attributes.id );
+
+	const { author, authors } = useSelect(
+		( select ) => {
+			const { getEntityRecord, getAuthors } = select( 'core' );
+			return {
+				author: getEntityRecord( 'root', 'user', postAuthorId ),
+				authors: getAuthors(),
+			};
+		},
+		[ postAuthorId ]
+	);
+
+	const { isSelected, fontSize, setFontSize } = props;
+	const {
+		TextColor,
+		BackgroundColor,
+		InspectorControlsColorPanel,
+		ColorPanel,
+	} = __experimentalUseColors(
+		[
+			{ name: 'textColor', property: 'color' },
+			{ name: 'backgroundColor', className: 'background-color' },
+		],
+		{
+			contrastCheckers: [
+				{
+					backgroundColor: true,
+					textColor: true,
+					fontSize: fontSize.size,
+				},
+			],
+			colorDetector: { targetRef: ref },
+			colorPanelProps: {
+				initialOpen: true,
+			},
+		},
+		[ fontSize.size ]
+	);
+
+	const { align, id, showAvatar, showBio, byline } = props.attributes;
+
+	const avatarSizes = [
+		{ value: 24, label: __( 'Small' ) },
+		{ value: 48, label: __( 'Medium' ) },
+		{ value: 96, label: __( 'Large' ) },
+	];
+
+	let avatarSize = DEFAULT_AVATAR_SIZE;
+	if ( !! props.attributes.avatarSize ) {
+		avatarSize = props.attributes.avatarSize;
+	}
+	useEffect( () => {
+		if ( author && ! props.attributes.id ) {
+			props.setAttributes( {
+				id: Number( author.id ),
+				name: author.name,
+				description: author.description,
+				avatarSize,
+				avatarUrl: author.avatar_urls[ avatarSize ],
+			} );
+		} else if ( authorId ) {
+			const authorFetch = apiFetch( {
+				path: '/wp/v2/users/' + authorId + '?context=edit',
+			} );
+			authorFetch.then( ( newAuthor ) => {
+				props.setAttributes( {
+					id: newAuthor.id,
+					name: newAuthor.name,
+					avatarUrl:
+						newAuthor.avatar_urls[ props.attributes.avatarSize ],
+					description: newAuthor.description,
+				} );
+			} );
+			return () => {
+				if ( authorFetch ) {
+					invoke( authorFetch, [ 'abort' ] );
+				}
+			};
+		}
+	}, [ authorId, author ] );
+
+	const blockClassNames = classnames( 'wp-block-post-author', {
+		[ fontSize.class ]: fontSize.class,
+	} );
```

Comment:
Doesn't need object notation -- `classnames` accepts strings, and ignores falsey args.

```suggestion
	const blockClassNames = classnames( 'wp-block-post-author', fontSize.class );
```

Reply:
Good catch, thanks!